providers/base/bin/wifi_nmcli_test.py

```diff
@@ -183,48 +172,59 @@ def perform_ping_test(interface):
     if target:
         count = 5
         result = ping(target, interface, count, 10)
-        if result["received"] == count:
-            return True
+        if result["received"] != count:
+            raise ValueError(
+                "{} packets expected but only {} received".format(
+                    count, result["received"]
+                )
+            )
 
-    return False
 
+@retry(max_attempts=5, delay=1)
+def wait_for_connected(interface, essid):
+    cmd = (
+        "nmcli -m tabular -t -f GENERAL.STATE,GENERAL.CONNECTION "
+        "d show {}".format(interface)
+    )
+    print_cmd(cmd)
+    output = sp.check_output(shlex.split(cmd))
+    print(output.decode(sys.stdout.encoding))
+    state, ssid = output.decode(sys.stdout.encoding).strip().splitlines()
```

Comment:
Minor: slightly easier to read
```suggestion
    output = sp.check_output(shlex.split(cmd), universal_newlines=True)
    print(output)
    state, ssid = output.strip().splitlines()
```